cmd/nknd/nknd.go

```diff
@@ -197,12 +199,46 @@ func nknMain(c *cli.Context) error {
 		}
 	}
 
+	if config.Parameters.Hostname == "" { // Skip query self extIP via set "HostName" in config.json
+		log.Info("Getting my IP address...")
+		var extIP string
+		if createMode { // There is no seed available in create mode, used ipify
+			extIP, err = ipify.GetIp()
+		} else {
+			extIP, err = AskMyIP(config.Parameters.SeedList)
+		}
+		if err != nil {
+			return err
+		}
+		log.Infof("My IP address is %s", extIP)
+		config.Parameters.Hostname = "64.64.245.149"
+	}
+
+	wssCertExists, httpsCertExists, prePreparedCert, _ := certs.CertExists()
```

Comment:
It's better to move L217-L233 into a single function call that returns two channel: wssCertReady and httpsCertReady